File: Util/RutaBase.cs
using System;
using System.IO;

namespace punto_de_venta_C_.Util
{
    public class RutaBase
    {
        private static string raiz = null;

        public static string getRaiz()
        {
            if (raiz != null) return raiz;
            
            raiz = AppDomain.CurrentDomain.BaseDirectory;
            return raiz;
        }

        public static string getImages()
        {
            string carpeta = Path.Combine(getRaiz(), "Images");
            if (!Directory.Exists(carpeta)) Directory.CreateDirectory(carpeta);
            return carpeta;
        }

        public static string getImagen(string nombre)
        {
            return Path.Combine(getImages(), nombre);
        }

        public static Microsoft.Maui.Controls.ImageSource getImagenSource(string ruta)
        {
            if (string.IsNullOrEmpty(ruta)) return null;
            try
            {
                string fullPath = ruta;
                if (!Path.IsPathRooted(fullPath))
                    fullPath = Path.Combine(getRaiz(), fullPath);
                fullPath = Path.GetFullPath(fullPath);
                if (File.Exists(fullPath))
                {
                    byte[] bytes = File.ReadAllBytes(fullPath);
                    return Microsoft.Maui.Controls.ImageSource.FromStream(() => new MemoryStream(bytes));
                }
            }
            catch (Exception)
            {
            }
            return null;
        }
    }
}

File: Util/RutaBase.py
import os
import sys

class RutaBase:
    _raiz = None

    @staticmethod
    def getRaiz():
        if RutaBase._raiz is not None:
            return RutaBase._raiz
        try:
            if getattr(sys, 'frozen', False):
                carpeta = os.path.dirname(sys.executable)
            else:
                carpeta = os.path.dirname(os.path.abspath(__file__))
            while carpeta:
                if os.path.exists(os.path.join(carpeta, "Images")) or os.path.exists(os.path.join(carpeta, "requirements.txt")):
                    RutaBase._raiz = carpeta
                    return RutaBase._raiz
                parent = os.path.dirname(carpeta)
                if parent == carpeta:
                    break
                carpeta = parent
        except Exception as e:
            print(e)
        RutaBase._raiz = os.getcwd()
        return RutaBase._raiz

    @staticmethod
    def getImages():
        carpeta = os.path.join(RutaBase.getRaiz(), "Images")
        if not os.path.exists(carpeta):
            os.makedirs(carpeta)
        return carpeta

    @staticmethod
    def getImagen(nombre):
        return os.path.join(RutaBase.getImages(), nombre)
